docs/strategy_analysis_example.md

```diff
@@ -250,3 +250,75 @@ fig.show()
 ```
 
 Feel free to submit an issue or Pull Request enhancing this document if you would like to share ideas on how to best analyze the data.
+
+## Analyse the buy/entry and sell/exit tags
+
+It can be helpful to understand how a strategy behaves according to the buy/entry tags used to
+mark up different buy conditions. You might want to see more complex statistics about each buy and
+sell condition above those provided by the default backtesting output. You may also want to
+determine indicator values on the signal candle that resulted in a trade opening.
+
+We first need to enable the exporting of trades from backtesting:
+
+```
+freqtrade backtesting -c <config.json> --timeframe <tf> --strategy <strategy_name> --timerange=<timerange> --export=trades --export-filename=user_data/backtest_results/<name>-<timerange>
+```
+
+To analyse the buy tags, we need to use the buy_reasons.py script in the `scripts/`
+folder. We need the signal candles for each opened trade so add the following option to your
+config file:
+
+```
+'backtest_signal_candle_export_enable': true,
+```
+
+This will tell freqtrade to output a pickled dictionary of strategy, pairs and corresponding
+DataFrame of the candles that resulted in buy signals. Depending on how many buys your strategy
+makes, this file may get quite large, so periodically check your `user_data/backtest_results`
+folder to delete old exports.
+
+Before running your next backtest, make sure you either delete your old backtest results or run
+backtesting with the `--cache none` option to make sure no cached results are used.
+
+If all goes well, you should now see a `backtest-result-{timestamp}_signals.pkl` file in the
+`user_data/backtest_results` folder.
+
+Now run the buy_reasons.py script, supplying a few options:
+
+```
+./scripts/buy_reasons.py -c <config.json> -s <strategy_name> -t <timerange> -g0,1,2,3,4
```

Comment:
If it's documented this way - then the script should also be included (otherwise we'd need some hint on how to get it there).

I'm however a bit concerned about throwing a new, 500+ line file into the scripts directory.
Scripts are usually not covered by tests - so it can break at any time, especially since it imports internal modules/packages from freqtrade.

Maybe you do have a version in another branch that does utilize the pickle-file already (and is cleaned up from the rest) - so we can actually see "how much" of that script is still left?

---

Instead of a script (we only really have the scripts folder for the rest client - as that intentionally doesn't depend on anything from freqtrade).
i think this potentially should be a subcommand (`freqtrade ...`) - which would make it testable, as well as (if done right) usable from within the jupyter notebook, or maybe even "inline" to backtesting via an additional flag.

I'm however fully aware that converting it to that may obviously be a bit more work than just dumping a script though ...

I would however see this as a secondary step (separate PR) ... to keep this one nice and small.

Reply:
I think i'll do as you suggest - splitting out the core functionality into a formal `freqtrade` command. I've done this for the notebook so it shouldn't be too much of a problem. We don't need the whole separate script.

Inline to backtesting would be fun! I'll leave that for another PR though.

Reply:
you might want to leave both of these for a separate PR - that way we can add this mostly "as is" for now - and have a followup PR that adds a new command ...

Reply:
OK, sounds good. I'll commit the buy_reasons.py script to the main FT scripts folder? Or simply link to my repo for now?

Reply:
link to your repo for now - that way we can avoid confusion i think